front/src/modules/simulationResult/types.ts

```diff
@@ -86,13 +86,14 @@ export type SpeedDistanceDiagramData = {
 export type ProjectionData = {
   path: TimetableItem['path'];
   projectedTrains: TrainSpaceTimeData[];
-  pathfinding: PathfindingResultSuccess;
-  geometry: PathProperties['geometry'];
+  pathfinding: PathfindingResultSuccess | undefined;
+  geometry: PathProperties['geometry'] | undefined;
   projectionLoaderData: {
     allTrainsProjected: boolean;
     totalTrains: number;
   };
   operationalPoints: PathProperties['operational_points'];
+  pathfindingHasFailed: boolean;
```

Comment:
nit: if the pathfinding has succeeded, then you know that the pathfinding and the geometry are defined
```
export type ProjectionData = { ... } & (
  | {
      pathfindingHasFailed: false;
      pathfinding: PathfindingResultSuccess;
      geometry: PathProperties['geometry'];
    }
  | {
      pathfindingHasFailed: true;
      pathfinding: undefined;
      geometry: undefined;
    }
)
```